Typescript import error (TS1444 and TS1371)

Describe a bug.

Steps to reproduce:
When I tried to build my app, the type check failed with following errors:

Expected behavior:
No build error, and develop build(npm dev run) don't show these errors.

Screenshots:
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
node_modules/@editorjs/editorjs/types/events/block/Base.ts:1:1 - error TS1371: This import is never used as a value and must use 'import type' because 'importsNotUsedAsValues' is set to 'error'.

1 import { BlockAPI } from '../../api';

node_modules/@editorjs/editorjs/types/events/block/Base.ts:1:10 - error TS1444: 'BlockAPI' is a type and must be imported using a type-only import when 'preserveValueImports' and 'isolatedModules' are both enabled.

1 import { BlockAPI } from '../../api';

node_modules/@editorjs/editorjs/types/events/block/BlockAdded.ts:1:1 - error TS1371: This import is never used as a value and must use 'import type' because 'importsNotUsedAsValues' is set to 'error'.

1 import { BlockMutationEventDetail } from './Base';

node_modules/@editorjs/editorjs/types/events/block/BlockAdded.ts:1:10 - error TS1444: 'BlockMutationEventDetail' is a type and must be imported using a type-only import when 'preserveValueImports' and 'isolatedModules' are both enabled.

1 import { BlockMutationEventDetail } from './Base';

node_modules/@editorjs/editorjs/types/events/block/BlockChanged.ts:1:1 - error TS1371: This import is never used as a value and must use 'import type' because 'importsNotUsedAsValues' is set to 'error'.

1 import { BlockMutationEventDetail } from './Base';

node_modules/@editorjs/editorjs/types/events/block/BlockChanged.ts:1:10 - error TS1444: 'BlockMutationEventDetail' is a type and must be imported using a type-only import when 'preserveValueImports' and 'isolatedModules' are both enabled.

1 import { BlockMutationEventDetail } from './Base';

node_modules/@editorjs/editorjs/types/events/block/BlockMoved.ts:1:1 - error TS1371: This import is never used as a value and must use 'import type' because 'importsNotUsedAsValues' is set to 'error'.

1 import { BlockMutationEventDetail } from './Base';

node_modules/@editorjs/editorjs/types/events/block/BlockMoved.ts:1:10 - error TS1444: 'BlockMutationEventDetail' is a type and must be imported using a type-only import when 'preserveValueImports' and 'isolatedModules' are both enabled.

1 import { BlockMutationEventDetail } from './Base';

node_modules/@editorjs/editorjs/types/events/block/BlockRemoved.ts:1:1 - error TS1371: This import is never used as a value and must use 'import type' because 'importsNotUsedAsValues' is set to 'error'.

1 import { BlockMutationEventDetail } from './Base';

node_modules/@editorjs/editorjs/types/events/block/BlockRemoved.ts:1:10 - error TS1444: 'BlockMutationEventDetail' is a type and must be imported using a type-only import when 'preserveValueImports' and 'isolatedModules' are both enabled.

1 import { BlockMutationEventDetail } from './Base';

node_modules/@editorjs/editorjs/types/events/block/index.ts:1:10 - error TS1444: 'BlockAddedEvent' is a type and must be imported using a type-only import when 'preserveValueImports' and 'isolatedModules' are both enabled.

1 import { BlockAddedEvent, BlockAddedMutationType } from './BlockAdded';

4 import { BlockRemovedEvent, BlockRemovedMutationType } from './BlockRemoved';
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Device, Browser, OS:
Desktop, Windows, Vite v4.3.9

Editor.js version:
2.27.0

Plugins you use with their versions:
Paragraph, AlignmentTune, NestedList, Undeline and some custom plugins (by me).

> Solution was simple for me, just add 'type' those error modules..
> like..
> import **type** { BlockMutationEventDetail } from './Base';
> but I'm not sure this solution is right.

Would you please reseach this issue?
Thank you in advance!

--- a/docs/CHANGELOG.md
+++ b/docs/CHANGELOG.md
@@ -6,6 +6,7 @@
 - `Fix` — Layout did not shrink when a large document cleared in Chrome
 - `Fix` — Multiple Tooltip elements creation fixed
 - `Fix` — When the focusing Block is out of the viewport, the page will be scrolled.
+- `Fix` - Resolve multple compiler errors from not importing as type
 
 ### 2.28.0
 

--- a/types/events/block/Base.ts
+++ b/types/events/block/Base.ts
@@ -1,4 +1,4 @@
-import { BlockAPI } from '../../api';
+import type { BlockAPI } from '../../api';
 
 /**
  * Details of CustomEvent fired on block mutation

--- a/types/events/block/BlockAdded.ts
+++ b/types/events/block/BlockAdded.ts
@@ -1,4 +1,4 @@
-import { BlockMutationEventDetail } from './Base';
+import type { BlockMutationEventDetail } from './Base';
 
 /**
  * Type name of CustomEvent related to block added event

--- a/types/events/block/BlockChanged.ts
+++ b/types/events/block/BlockChanged.ts
@@ -1,4 +1,4 @@
-import { BlockMutationEventDetail } from './Base';
+import type { BlockMutationEventDetail } from './Base';
 
 /**
  * Type name of CustomEvent related to block changed event

--- a/types/events/block/BlockMoved.ts
+++ b/types/events/block/BlockMoved.ts
@@ -1,4 +1,4 @@
-import { BlockMutationEventDetail } from './Base';
+import type { BlockMutationEventDetail } from './Base';
 
 /**
  * Type name of CustomEvent related to block moved event

--- a/types/events/block/BlockRemoved.ts
+++ b/types/events/block/BlockRemoved.ts
@@ -1,4 +1,4 @@
-import { BlockMutationEventDetail } from './Base';
+import type { BlockMutationEventDetail } from './Base';
 
 /**
  * Type name of CustomEvent related to block removed event

--- a/types/events/block/index.ts
+++ b/types/events/block/index.ts
@@ -1,7 +1,7 @@
-import { BlockAddedEvent, BlockAddedMutationType } from './BlockAdded';
-import { BlockChangedEvent, BlockChangedMutationType } from './BlockChanged';
-import { BlockMovedEvent, BlockMovedMutationType } from './BlockMoved';
-import { BlockRemovedEvent, BlockRemovedMutationType } from './BlockRemoved';
+import { type BlockAddedEvent, BlockAddedMutationType } from './BlockAdded';
+import { type BlockChangedEvent, BlockChangedMutationType } from './BlockChanged';
+import { type BlockMovedEvent, BlockMovedMutationType } from './BlockMoved';
+import { type BlockRemovedEvent, BlockRemovedMutationType } from './BlockRemoved';
 
 /**
  * Map for Custom Events related to block mutation types

